### PR TITLE
Add `enable_builtin_logging` to `azurerm_function_app`

### LINKED ISSUE #1979: Make AzureWebJobsDashboard optional for Function Apps
Hey all,

Based on the documentation here https://docs.microsoft.com/en-us/azure/azure-functions/functions-monitoring#disable-built-in-logging it would be nice for the AzureWebJobsDashboard app setting to be optional.

> If you enable Application Insights, we recommend that you disable the built-in logging that uses Azure storage. The built-in logging is useful for testing with light workloads but is not intended for high-load production use

> To disable built-in logging, delete the AzureWebJobsDashboard app setting

I'm note sure what the ideal API would look like, but something along the lines of the following might work (with disable_inbuilt_logging defaulting to false to keep backwards compatibility)?

```hcl
resource "azurerm_function_app" "fa" {
  name                                   = "scott-fnapp-dev-eun-func"
  location                               = "${azurerm_resource_group.rg.location}"
  resource_group_name        = "${azurerm_resource_group.rg.name}"
  app_service_plan_id            = "${azurerm_app_service_plan.sp.id}"
  storage_connection_string = "${azurerm_storage_account.sa.primary_connection_string}"
  disable_inbuilt_logging      = true
}
```

Cheers!

### PR DESCRIPTION
**Pull request by JunyiYi**
This PR fulfills the request in #1979.
The method used in the implementation is documented here: https://docs.microsoft.com/en-us/azure/azure-functions/functions-monitoring#disable-built-in-logging.

(fixes #1979)

### CODE REVIEW

File: azurerm/resource_arm_function_app.go [hunk 1]
```diff
@@ -66,6 +66,12 @@ func resourceArmFunctionApp() *schema.Resource {
 				Optional: true,
 			},
 
+			"built_in_logging_enabled": {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app.go]: we've got previous examples of using `builtin` vs `built_in` - such that I think this'd make more sense as:

```suggestion
			"enable_builtin_logging": {
```

File: azurerm/resource_arm_function_app.go [hunk 2]
```diff
@@ -452,6 +458,9 @@ func resourceArmFunctionAppRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("storage_connection_string", appSettings["AzureWebJobsStorage"])
 	d.Set("version", appSettings["FUNCTIONS_EXTENSION_VERSION"])
 
+	dashboard, ok := appSettings["AzureWebJobsDashboard"]
+	d.Set("built_in_logging_enabled", ok && dashboard != "")
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app.go]: which'd make this:

```suggestion
	d.Set("enable_builtin_logging", ok && dashboard != "")
```

File: azurerm/resource_arm_function_app.go [hunk 3]
```diff
@@ -524,11 +533,17 @@ func getBasicFunctionAppAppSettings(d *schema.ResourceData, appServiceTier strin
 	contentShare := strings.ToLower(d.Get("name").(string)) + "-content"
 
 	basicSettings := []web.NameValuePair{
-		{Name: &dashboardPropName, Value: &storageConnection},
 		{Name: &storagePropName, Value: &storageConnection},
 		{Name: &functionVersionPropName, Value: &functionVersion},
 	}
 
+	if d.Get("built_in_logging_enabled").(bool) {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app.go]: which'd make this:

```suggestion
	if d.Get("enable_builtin_logging").(bool) {
```

Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app.go]: which'd make this:

```suggestion
	if d.Get("enable_builtin_logging").(bool) {
```

File: azurerm/resource_arm_function_app_test.go [hunk 4]
```diff
@@ -455,6 +455,73 @@ func TestAccAzureRMFunctionApp_updateIdentity(t *testing.T) {
...
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMFunctionAppDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMFunctionAppExists(resourceName),
+					testCheckAzureRMFunctionAppHasNoContentShare(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "built_in_logging_enabled", "false"),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app_test.go]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "enable_builtin_logging", "false"),
```

File: azurerm/resource_arm_function_app_test.go [hunk 5]
```diff
@@ -455,6 +455,73 @@ func TestAccAzureRMFunctionApp_updateIdentity(t *testing.T) {
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMFunctionAppDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: enabledConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAppServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "built_in_logging_enabled", "true"),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app_test.go]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "enable_builtin_logging", "true"),
```

File: azurerm/resource_arm_function_app_test.go [hunk 6]
```diff
@@ -455,6 +455,73 @@ func TestAccAzureRMFunctionApp_updateIdentity(t *testing.T) {
...
+				),
+			},
+			{
+				Config: disabledConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMFunctionAppExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "built_in_logging_enabled", "false"),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app_test.go]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "enable_builtin_logging", "false"),
```

File: azurerm/resource_arm_function_app_test.go [hunk 7]
```diff
@@ -455,6 +455,73 @@ func TestAccAzureRMFunctionApp_updateIdentity(t *testing.T) {
...
+				),
+			},
+			{
+				Config: enabledConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAppServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "built_in_logging_enabled", "true"),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app_test.go]:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "enable_builtin_logging", "true"),
```

File: azurerm/resource_arm_function_app_test.go [hunk 8]
```diff
@@ -1126,3 +1193,38 @@ resource "azurerm_function_app" "test" {
...
+	}
+}
+
+resource "azurerm_function_app" "test" {
+	name                      = "acctest-%[1]d-func"
+	location                  = "${azurerm_resource_group.test.location}"
+	resource_group_name       = "${azurerm_resource_group.test.name}"
+	app_service_plan_id       = "${azurerm_app_service_plan.test.id}"
+	storage_connection_string = "${azurerm_storage_account.test.primary_connection_string}"
+	built_in_logging_enabled  = false
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app_test.go]:
```suggestion
	enable_builtin_logging  = false
```

File: azurerm/resource_arm_function_app_test.go [hunk 9]
```diff
@@ -1126,3 +1193,38 @@ resource "azurerm_function_app" "test" {
...
+}`, rInt, location, storage)
+}
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_function_app_test.go]: can we fix the linting here?

File: azurerm/resource_arm_function_app.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_function_app.go]: can we also update the documentation to include this new field?

File: azurerm/resource_arm_function_app_test.go (hunk 9 above)
Comment (JunyiYi) [reply azurerm/resource_arm_function_app_test.go]: I did not see any linting errors here?

**Comment by katbyte** [comment]
Tests pass:
screen shot 2018-11-08 at 17 50 06

**Pull request merged**